.github/workflows/main.yml

```diff
@@ -0,0 +1,41 @@
+name: build
+on:
+  # run on push but only for the master branch
+  push:
+    branches:
+      - master
```

Comment:
Would it be useful to just run on every branch? Good to know if feature branches are passing tests too

Reply:
We can run it on every branch. It works well if everyone forks the main repo and sends PRs. But if we work off of upstream repos (like this PR is doing) you end up double running CI on PRs like so:

<img width="1000" alt="Screen Shot 2020-12-07 at 4 17 27 PM" src="https://user-images.githubusercontent.com/534034/101412312-fe2d2f00-38a7-11eb-99d8-1f41df9df9fb.png">

If we limit it to just the main branch for "push" with the following code then we can continue working off upstream and not have this problem:

```
  # run on push but only for the master branch
  push:
    branches:
      - master
```

I'm also paranoid about kicking off too many jobs with GitHub Actions and causing things to queue up and take a long time. So I've been on the conservative side of only running CI when absolutely necessary.

Reply:
@bluepnume  I'd like to leave in this "push but only for the master branch" setting. Let me know if you have any questions with the explanation above.